python/pyspark/sql/connect/protobuf/functions.py

```diff
@@ -0,0 +1,164 @@
+    binaryDescriptorSet: Optional[bytes] = None,
+) -> Column:
+    binary_proto = None
+    if binaryDescriptorSet is not None:
+        binary_proto = binaryDescriptorSet
+    elif descFilePath is not None:
+        binary_proto = _read_descriptor_set_file(descFilePath)
+
+    if binary_proto is not None:
+        if options is None:
+            return _invoke_function(
+                "from_protobuf", _to_col(data), lit(messageName), lit(binary_proto)
+            )
+        else:
+            return _invoke_function(
+                "from_protobuf",
+                _to_col(data),
+                lit(messageName),
+                lit(binary_proto),
+                _options_to_col(options),
+            )
+    else:
+        if options is None:
+            return _invoke_function("from_protobuf", _to_col(data), lit(messageName))
+        else:
+            return _invoke_function(
+                "from_protobuf", _to_col(data), lit(messageName), _options_to_col(options)
+            )
+
+
+from_protobuf.__doc__ = PyProtobufFunctions.from_protobuf.__doc__
+
+
+def to_protobuf(
+    data: "ColumnOrName",
+    messageName: str,
+    descFilePath: Optional[str] = None,
+    options: Optional[Dict[str, str]] = None,
+    binaryDescriptorSet: Optional[bytes] = None,
+) -> Column:
+    binary_proto = None
+    if binaryDescriptorSet is not None:
+        binary_proto = binaryDescriptorSet
+    elif descFilePath is not None:
+        binary_proto = _read_descriptor_set_file(descFilePath)
+
+    if binary_proto is not None:
+        if options is None:
+            return _invoke_function(
+                "to_protobuf", _to_col(data), lit(messageName), lit(binary_proto)
+            )
+        else:
+            return _invoke_function(
+                "to_protobuf",
+                _to_col(data),
+                lit(messageName),
+                lit(binary_proto),
+                _options_to_col(options),
+            )
+    else:
+        if options is None:
+            return _invoke_function("to_protobuf", _to_col(data), lit(messageName))
+        else:
+            return _invoke_function(
+                "to_protobuf", _to_col(data), lit(messageName), _options_to_col(options)
+            )
+
+
+to_protobuf.__doc__ = PyProtobufFunctions.to_protobuf.__doc__
+
+
+def _read_descriptor_set_file(filePath: str) -> bytes:
+    with open(filePath, "rb") as f:
+        return f.read()
+
+
+def _test() -> None:
+    import os
+    import sys
+    from pyspark.testing.utils import search_jar
+
+    protobuf_jar = search_jar("connector/protobuf", "spark-protobuf-assembly-", "spark-protobuf")
+    if protobuf_jar is None:
+        print(
+            "Skipping all Protobuf Python tests as the optional Protobuf project was "
+            "not compiled into a JAR. To run these tests, "
+            "you need to build Spark with 'build/sbt package' or "
+            "'build/mvn package' before running this test."
+        )
+        sys.exit(0)
```

Comment:
This should not fail like this right?

Reply:
This part is just copied from the original [protobuf functions](https://github.com/apache/spark/blob/master/python/pyspark/sql/protobuf/functions.py#L295-L302). Should we do it differently here?